src/wallet/wallet.cpp

```diff
@@ -3252,6 +3250,7 @@ bool ReserveDestination::GetReservedDestination(const OutputType type, CTxDestin
         return false;
     }
 
+    m_spk_man->TopUp();
```

Comment:
In commit "Key pool: Move TopUp call" (8b9a1ed9fb5ef64385d3881b1423059e4daaef13)

Since aside from the call in `GetNewChangeDestination` this code is also called from `CWallet::CreateTransaction`, I'm wondering if this is a change in behavior. Is topup already called when creating transactions? Wondering if this the "unsollicited keypool topup" refered to by https://github.com/bitcoin/bitcoin/pull/16341#issuecomment-510903836

Reply:
This is not a change in behavior because `TopUp` is called in `LegacyScriptPubKeyMan::ReserveKeyFromKeyPool`

Reply:
> So if it's already called then why do we need to add a call here?

It's a belt-and-suspenders as not all ScriptPubKeyMans are guaranteed to do this.

I've moved it inside the block.